bundles/org.dataflowanalysis.analysis.pcm/src/org/dataflowanalysis/analysis/pcm/PCMDataFlowConfidentialityAnalysis.java

```diff
@@ -4,94 +4,75 @@
-		PCMResourceProvider resourceProvider = (PCMResourceProvider) this.resourceProvider;
-		ActionSequenceFinder sequenceFinder = new PCMActionSequenceFinder(resourceProvider.getUsageModel());
-        return sequenceFinder.findAllSequences().parallelStream()
-        		.map(ActionSequence.class::cast)
-        		.collect(Collectors.toList());
-	}
-
-	@Override
-	public List<ActionSequence> evaluateDataFlows(List<ActionSequence> sequences) {
-		List<PCMActionSequence> actionSequences = sequences.parallelStream()
-    			.map(PCMActionSequence.class::cast)
-    			.collect(Collectors.toList());
-    	return actionSequences.parallelStream()
-    	          .map(it -> it.evaluateDataFlow(this.nodeCharacteristicsCalculator, this.dataCharacteristicsCalculatorFactory))
-    	          .toList();
-	}
-
-	@Override
-	public List<AbstractActionSequenceElement<?>> queryDataFlow(ActionSequence sequence,
-			Predicate<? super AbstractActionSequenceElement<?>> condition) {
-		return sequence.getElements()
-	            .parallelStream()
-	            .filter(condition)
-	            .toList();
-	}
-	
-	@Override
-    public boolean initializeAnalysis() {
+    private static final String PLUGIN_PATH = "org.dataflowanalysis.analysis.pcm";
+    private final Logger logger;
+
+    protected final PCMResourceProvider resourceProvider;
+
+    protected final String modelProjectName;
+    protected final Optional<Class<? extends Plugin>> modelProjectActivator;
+
+    protected List<PCMDataDictionary> dataDictionaries;
+
+    /**
+     * Creates a new instance of a data flow analysis with the given parameters
+     * @param resourceProvider Resource provider that provides the required resources to the analysis
+     * @param modelProjectName Name of the modelling project
+     * @param modelProjectActivator Plugin class of the analysis
+     */
+    public PCMDataFlowConfidentialityAnalysis(PCMResourceProvider resourceProvider, String modelProjectName,
+            Optional<Class<? extends Plugin>> modelProjectActivator) {
+        this.resourceProvider = resourceProvider;
+        this.logger = Logger.getLogger(PCMDataFlowConfidentialityAnalysis.class);
+        this.modelProjectName = modelProjectName;
+        this.modelProjectActivator = modelProjectActivator;
+    }
+
+    @Override
+    public PCMFlowGraph findFlowGraph() {
+        return new PCMFlowGraph(this.resourceProvider);
+    }
+
+    @Override
+    public PCMFlowGraph evaluateFlowGraph(FlowGraph flowGraph) {
+        if (!(flowGraph instanceof PCMFlowGraph pcmFlowGraph)) {
+            logger.error("Cannot evaluate non-pcm flow graph!");
+            throw new IllegalArgumentException();
+        }
+        return pcmFlowGraph.evaluate();
+    }
+
+    @Override
+    public List<? extends AbstractVertex<?>> queryDataFlow(AbstractPartialFlowGraph sequence, Predicate<? super AbstractVertex<?>> condition) {
+        return sequence.getVertices().parallelStream().filter(condition).toList();
```

Comment:
I opt to remove the parallel stream here as we don't need the potential performance gain in an O(n) method and this kills the ordering and determinism

Reply:
Changed to an sequential stream in 83528798a0d7e69387cc8132b1a96b6f3fb6d786